server/src/services/poaching-incident.service.ts

```diff
@@ -14,32 +27,94 @@ export class PoachingIncidentService {
     description: string,
   ): Promise<boolean> {
     const con = await this.databaseService.getConnection();
-    const poachingIncident = new PoachingIncident();
+    //const poachingIncident = new PoachingIncident();
+
+    const reserve = await con.getRepository(ReserveConfiguration).findOne({});
+
+    console.log(reserve.reserveName);
+
+    const mapData = await this.mapUpdater.getMapFeatures(
+      reserve.reserveName,
+    );
+
+    const bounds = bbox(mapData.reserve);
+    console.time('feature searchers');
+    const featureSearchers: { [s: string]: GeoSearchSet } = Object.keys(
+      mapData.features,
+    ).reduce((searchers, featureName) => {
+      searchers[featureName] = this.geo.createFastSearchDataset(
+        mapData.features[featureName],
+      );
+      return searchers;
+    }, {});
+    console.timeEnd('feature searchers');
+
+    console.timeEnd('get map data');
+
+    // if not parse then it gives not a number
+    const latitude = parseFloat(lat.toString());
+    const longitude = parseFloat(lon.toString());
+    const locationBounds = [
+      longitude - LOCATION_BIAS, // left
+      latitude - LOCATION_BIAS, // bottom
+      longitude + LOCATION_BIAS, // right
+      latitude + LOCATION_BIAS, // top
+    ];
+
+    const altitudeInfo = await this.altitude.getAltitude(
+      locationBounds,
+      bounds,
+    );
 
     try {
       const poachingIncidentType = await con
         .getRepository(PoachingIncidentType)
         .findOne({ type: pType });
 
-        if(poachingIncidentType == undefined) {
-          console.log("The type of incident does not exist.")
-          return false;
-    } else {
-      poachingIncident.timestamp = new Date();
-      poachingIncident.longitude = lon;
-      poachingIncident.latitude = lat;
-      poachingIncident.description = description
-      poachingIncident.type = poachingIncidentType;
-      // tslint:disable-next-line:no-console
-      const addedPoachingIncident = await con
-        .getRepository(PoachingIncident)
-        .save(poachingIncident);
-      console.log(
-        'Saved a new poaching incident with id: ' + poachingIncident.id,
-      );
+      if (poachingIncidentType == undefined) {
+        console.log('The type of incident does not exist.');
+        return false;
+      } else {
+        const date = new Date();
 
-      return addedPoachingIncident != null;
-    }
+        const poachingIncident: PoachingIncident = {
+          timestamp : date,
+          longitude: lon,
+         latitude: lat,
+          description: description,
+          type: poachingIncidentType,
+          month: date.getMonth() + 1,
+          time: date.getHours() * 60 + date.getMinutes(),
+          CoordinateData: "distanceToDams: " + featureSearchers.dams.getNearest(lon, lat).distance 
```

Comment:
I think it would be better to use JSON.stringify here

```
JSON.stringify({
 distanceToDams: featureSearchers.dams.getNearest(lon, lat).distance,
 ...
})
```

Reply:
And then use JSON.parse when decoding it

Reply:
Will do it as suggested